### ISSUE TITLE
[a11y]: Screen reader still fail to announce batch toolbar in v11

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#17756]
### Package

@carbon/react

### Browser

Edge

### Operating System

Windows

### Package version

Current storybook

### React version

@carbon/react@1.68.0

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

The same issue described in #5242 -- although I think they meant _row_ not column...

I fiddled around and if one uses aria-live="polite" it is relatively easy to get some of the batch action buttons to be announced. It's a bit less easy to get the status message announced, partly because the "O rows selected" is already populated in the aria-hidden div. It would probably make more sense to have this an empty string. The populated text should then be announced. OR, you could try not having it aria-hidden to begin with and literally a blank div, then add the text, which would then make it available.

### WCAG 2.1 Violation

4.1.3

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

### Steps to reproduce

1. Open https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default
2. Start JAWS (or other SR)
3. navigate to any table row and select with spacebar. The batch row will appear, however, the status information (e.g., "6 rows selected") will not be announced.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
